gprofiler/metadata/application_identifiers.py

```diff
@@ -14,14 +14,6 @@
 from gprofiler.log import get_logger_adapter
 from gprofiler.metadata.base_application_identifier import _ApplicationIdentifier
 from gprofiler.metadata.enrichment import EnrichmentOptions
-from gprofiler.platform import is_linux
-from gprofiler.profilers.java import JattachJcmdRunner
```

Comment:
The problematic import is `JattachJcmdRunner`:
```
Traceback (most recent call last):
  File "pyi_build.py", line 11, in <module>
    from gprofiler.main import main
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "PyInstaller\loader\pyimod03_importers.py", line 49[5](https://github.com/Granulate/gprofiler/actions/runs/4252348474/jobs/7395819982#step:4:6), in exec_module
  File "gprofiler\main.py", line 44, in <module>
    from gprofiler.metadata.application_identifiers import ApplicationIdentifiers
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 98[6](https://github.com/Granulate/gprofiler/actions/runs/4252348474/jobs/7395819982#step:4:7), in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "PyInstaller\loader\pyimod03_importers.py", line 495, in exec_module
  File "gprofiler\metadata\application_identifiers.py", line 18, in <module>
    from gprofiler.profilers.java import JattachJcmdRunner
  File "<frozen importlib._bootstrap>", line 100[7](https://github.com/Granulate/gprofiler/actions/runs/4252348474/jobs/7395819982#step:4:8), in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "PyInstaller\loader\pyimod03_importers.py", line 495, in exec_module
  File "gprofiler\profilers\java.py", line 64, in <module>
    from gprofiler.kernel_messages import get_kernel_messages_provider
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 9[8](https://github.com/Granulate/gprofiler/actions/runs/4252348474/jobs/7395819982#step:4:9)6, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "PyInstaller\loader\pyimod03_importers.py", line 495, in exec_module
  File "gprofiler\kernel_messages.py", line 5, in <module>
    from granulate_utils.linux.kernel_messages import (
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "PyInstaller\loader\pyimod03_importers.py", line 4[9](https://github.com/Granulate/gprofiler/actions/runs/4252348474/jobs/7395819982#step:4:10)5, in exec_module
  File "granulate_utils\linux\kernel_messages.py", line 30, in <module>
  File "granulate_utils\linux\__init__.py", line [11](https://github.com/Granulate/gprofiler/actions/runs/4252348474/jobs/7395819982#step:4:12), in get_kernel_release
AttributeError: module 'os' has no attribute 'uname'
[46[36](https://github.com/Granulate/gprofiler/actions/runs/4252348474/jobs/7395819982#step:4:37)] Failed to execute script 'pyi_build' due to unhandled exception!
```

(from https://github.com/Granulate/gprofiler/actions/runs/4252348474/jobs/7395819982)

Let's put it under the `if is_linux()`?

The name is only required for typing purposes - this can be handled by adding `from __future__ import annotations`, which allows the annotation `jattach_jcmd_runner: JattachJcmdRunner` to be undefined, and when mypy runs it will be able to get the correct import.

This is a smaller diff so I suggest